Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py

```diff
@@ -811,8 +840,174 @@ def delete_file_command(client: MsGraphClient, args):
     return human_readable, text  # == raw response
 
 
+def get_site_id_from_site_name(client: MsGraphClient, site_name: str | None) -> str:
+    """
+    Retrieves the SharePoint site ID based on either the provided site ID or site name.
+
+    Args:
+        client (MsGraphClient): An instance of the Microsoft Graph client.
+        site_name (Optional[str]): The name of the SharePoint site.
+
+    Returns:
+        str: The SharePoint site ID.
+
+    Raises:
+        DemistoException: If neither site_id nor site_name is provided,
+        or if the specified site_name does not correspond to a valid SharePoint site.
+    """
+    if site_name:
+        site_details = client.list_sharepoint_sites(site_name)["value"]
+        if site_details:
+            return site_details[0]["id"]
+        else:
+            raise DemistoException(f"Site '{site_name}' not found. Please provide a valid site name.")
+
+    raise DemistoException("Either 'site_id' or 'site_name' parameter is required.")
+
+
+def _parse_permission(permission: dict):
+    identities: list[dict[str, dict[str, str]]] = permission.get("grantedToIdentitiesV2", [{}])
+    return {
+        "ID": permission.get("id"),
+        "Roles": permission.get("roles"),
+        "Application Name": [identity.get("application", {}).get("displayName") for identity in identities],
+        "Application ID": [identity.get("application", {}).get("id") for identity in identities],
+    }
+
+
+def list_site_permissions_command(client: MsGraphClient, args: dict[str, str]) -> CommandResults:
+    """Lists permissions for a SharePoint site.
+
+    Args:
+        client (MsGraphClient): The Microsoft Graph client.
+        args (dict): The command arguments.
+            permission_id (Optional): The unique identifier for the permission.
+            limit (Optional): Number of items to return. Default is 50.
+            all_results (Optional): Return all results. Default is False.
+            site_name (Optional): The display name of the site. Either site_id or site_name is required.
+            site_id (Optional): The unique identifier for the site. Either site_id or site_name is required.
+
+    Returns:
+        CommandResults: The command results with outputs and readable output.
+    """
+    permission_id = args.get("permission_id")
+    limit = arg_to_number(args.get("limit", 50))
+    all_results = argToBoolean(args.get("all_results", False))
+    site_id = args.get("site_id") or get_site_id_from_site_name(client, args.get("site_name"))
+    results = client.list_site_permissions(site_id, permission_id)
+    if permission_id:
+        return CommandResults(
+            outputs_prefix="MsGraphFiles.SitePermission",
+            outputs_key_field="id",
+            outputs=results,
+            readable_output=tableToMarkdown(
+                name="Site Permission",
+                t=_parse_permission(results)
+            )
+        )
+
+    list_permissions = results.get("value", [])
```

Comment:
Do we always get a list even when there is only one result? If not we should consider casting to list so the context will look correct.

Reply:
Yes, we are going to get `list` always.